packages/react-query/src/shared/proxy/utilsProxy.ts

```diff
@@ -107,7 +107,7 @@ type DecorateProcedure<
       predicate?: (
         query: Query<
           inferProcedureInput<TProcedure>,
-          TRPCClientError<TRouter>,
+          TRPCClientError<TProcedure>,
```

Comment:
So this was encoding whole router information into the type defs for a single procedure, which broke this one polymorphism corner case. 

Not sure what impacts this will really have though

Reply:
`TRPCClientError<TProcedure>` is what we should be using everywhere 👌

Reply:
Computer says 🤔

<img width="583" alt="image" src="https://user-images.githubusercontent.com/8896153/230459192-26acb5d6-c672-483e-988d-9a6063bbc21a.png">